File: Kapitel14/14.5/Konto/Konto.cpp
/* Programmname: Konto
 * Fachkonzept-Klasse: Konto
 * Aufgabe: konkrete Oberklasse Konto
 * Dateiname: Konto.cpp
 */
#include "Konto.h"

Konto::Konto(int nummer, double ersteZahlung) : // Initialisierungsliste
                                                kontonr(nummer), kontostand(ersteZahlung)
{
}

// Schreibende Operationen
void Konto::buchen(double betrag)
{
    kontostand = kontostand + betrag;
}

// Lesende Operationen
double Konto::getKontostand() const
{
    return kontostand;
}

File: Kapitel15/15.4/Konto/Konto.cs
/* Programmname: Konto
 * UI-Klasse: KontoUI
 * Aufgabe: Konten verwalten
 * Eingabe von Beträgen und Kontoart
 * Ausgabe des aktuellen Kontostands
 */
public class Konto
{
    //Attribute
    protected int kontonr;
    protected double kontostand;

    // Konstruktor
    public Konto(int kontonr, double ersteZahlung)
    {
        this.kontonr = kontonr;
        kontostand = ersteZahlung;
    }

    //Schreibende Operationen
    public virtual void buchen(double betrag)
    {
        kontostand = kontostand + betrag;
    }

    //Lesende Operationen
    public double getKontostand()
    {
        return kontostand;
    }
}